File: solutions/169.majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const majorityElement = nums => {
  let hash = {}

  let len = nums.length
  for (let i = 0; i < len; i++) {
    let item = nums[i]
    if (item in hash) {
      hash[item]++
    } else {
      hash[item] = 1
    }

    if (hash[item] > len / 2) return item
  }
}

File: solutions/169.majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        target = len(nums) / 2
        m = {}
        for n in nums:
            m[n] = m.get(n, 0) + 1
            if m[n] > target:
                return n